src/MultiAgentReinforcementLearning/src/policies/maddpg.jl

```diff
@@ -0,0 +1,157 @@
+export MADDPGManager
+using BSON
+using Flux
+using Statistics
+
+"""
+    MADDPGManager(; agents::Dict{<:Any, <:Agent}, args...)
+Multi-agent Deep Deterministic Policy Gradient(MADDPG) implemented in Julia. By default, `MADDPGManager` uses for simultaneous environments with [continuous action space](https://spinningup.openai.com/en/latest/spinningup/rl_intro.html#stochastic-policies).
+See the paper https://arxiv.org/abs/1706.02275 for more details.
+
+# Keyword arguments
+- `agents::Dict{<:Any, <:Agent}`, here each agent collects its own information. While updating the policy, each **critic** will assemble all agents' 
+  trajectory to update its own network. **Note that** here the policy of the `Agent` should be `DDPGPolicy` wrapped by `NamedPolicy`, see the relative 
+  experiment([`MADDPG_KuhnPoker`](https://juliareinforcementlearning.org/docs/experiments/experiments/Policy%20Gradient/JuliaRL_MADDPG_KuhnPoker/#JuliaRL\\_MADDPG\\_KuhnPoker) or [`MADDPG_SpeakerListener`](https://juliareinforcementlearning.org/docs/experiments/experiments/Policy%20Gradient/JuliaRL_MADDPG_SpeakerListener/#JuliaRL\\_MADDPG\\_SpeakerListener)) for references.
+- `traces`, set to `SARTS` if you are apply to an environment of `MINIMAL_ACTION_SET`, or `SLARTSL` if you are to apply to an environment of `FULL_ACTION_SET`.
+- `batch_size::Int`
+- `update_freq::Int`
+- `update_step::Int`, count the step.
+- `rng::AbstractRNG`.
+"""
+mutable struct MADDPGManager{T <: AbstractPolicy} <: AbstractPolicy
+    agents::Dict{<:Any, T}
+    batch_size::Int
+    update_freq::Int
+    update_step::Int
+    rng::AbstractRNG
```

Comment:
Same thing here, not sure what you see if you run `JET.jl` in terms of type stability, but it might make sense to use a param in `MADDPGManager` for the rng type.

Reply:
Actually, the range parameter is not needed here, since the samplers of the trajectories are doing the job. Previously, every policy needed to take care of the sampling on its own. This is not necessary anymore.